modules/public/views.py

```diff
@@ -284,6 +325,35 @@ def external(attrs, new=False):
 
         formatted_paragraphs.append(paragraph.replace('\n', '<br />'))
 
+    status_code = 200
+    country_data = []
+    try:
+        metrics_query_json = [
+            {
+                "metric_name": "installed_base_by_country_percent",
+                "snap_id": details['snap_id'],
+                "start": week_ago.strftime('%Y-%m-%d'),
+                "end": today.strftime('%Y-%m-%d')
+            }
+        ]
+
+        metrics_response = api.get_public_metrics(
+            snap_name,
+            metrics_query_json
+        )
+
+        users_by_country = normalize_metrics(
+            metrics_response[0]['series']
+        )
+        country_data = build_country_info(users_by_country)
+    except api.ApiResponseError as api_error_exception:
+        if hasattr(api_error_exception, 'errors'):
+            errors = api_error_exception.errors
+        else:
+            errors.append(str(api_error_exception))
+
+        status_code = api_error_exception.status_code
```

Comment:
We shouldn't just return the `status_code` from the API. This is how the *API* tells *us* what went wrong on their end, but that's not necessarily the same thing that we want to tell our client.

Basically, we need to decide if we consider ourselves a "gateway" to the API, in the HTTP sense. If we are then most API errors should be communicated as a "502 Bad Gateway" error, except for the specific case of an API timeout, in which case we should send a "504 Gateway Timeout" error.

If we don't consider ourselves a "gateway" in the technical sense, then all errors should simply be communicated as "500 Internal Server Error".

[RFC2616 sec1](https://www.w3.org/Protocols/rfc2616/rfc2616-sec1.html#sec1) describes a "gateway" as:

> A server which acts as an intermediary for some other server. Unlike a proxy, a gateway receives requests as if it were the origin server for the requested resource; the requesting client may not be aware that it is communicating with a gateway.

To me that sounds a lot like what we're doing here, so I would suggest we should use `502` in most cases, and `504` in the case of timeouts.